intermediate_source/torch_export_tutorial.py

```diff
@@ -0,0 +1,462 @@
+        return torch.sin(x)
+    return torch.cos(x)
+
+import traceback as tb
+try:
+    export(bad1, (torch.randn(3, 3),))
+except Exception:
+    tb.print_exc()
+
+######################################################################
+# - accessing tensor data with ``.data``
+
+def bad2(x):
+    x.data[0, 0] = 3
+    return x
+
+try:
+    export(bad2, (torch.randn(3, 3),))
+except Exception:
+    tb.print_exc()
+
+
+######################################################################
+# - calling unsupported functions (e.g. many builtins)
+
+def bad3(x):
+    x = x + 1
+    return x + id(x)
+
+try:
+    export(bad3, (torch.randn(3, 3),))
+except Exception:
+    tb.print_exc()
+
+######################################################################
+# - unsupported Python language features (e.g. throwing exceptions, match statements)
+
+def bad4(x):
+    try:
+        x = x + 1
+        raise RuntimeError("bad")
+    except:
+        x = x + 2
+    return x
+
+try:
+    export(bad4, (torch.randn(3, 3),))
+except Exception:
+    tb.print_exc()
+
+######################################################################
+# Control Flow Ops
+# ----------------
+# ``torch.export`` actually does support data-dependent control flow.
+# But these need to be expressed using control flow ops. For example,
+# we can fix the control flow example above using the ``cond`` op, like so:
+
+from functorch.experimental import control_flow
+
+def bad1_fixed(x):
+    def true_fn(x):
+        return torch.sin(x)
+    def false_fn(x):
+        return torch.cos(x)
+    return control_flow.cond(x.sum() > 0, true_fn, false_fn, [x])
+
+exported_bad1_fixed = export(bad1_fixed, (torch.randn(3, 3),))
+print(exported_bad1_fixed(torch.ones(3, 3)))
+print(exported_bad1_fixed(-torch.ones(3, 3)))
+
+######################################################################
+# There are some limitations one should be aware of:
+# - The predicate (i.e. ``x.sum() > 0``) must result in a boolean or a single-element tensor.
+# - The operands (i.e. ``[x]``) must be tensors.
+# - The branch function (i.e. ``true_fn`` and ``false_fn``) signature must match with the
+# operands and they must both return a single tensor with the same metadata (e.g. dtype, shape, etc.)
+# - Branch functions cannot mutate inputs or globals
+# - Branch functions cannot access closure variables, except for ``self`` if the function is
+# defined in the scope of a method.
+
+######################################################################
+# We can also use ``map``, which applies a function across the first dimension
+# of the first tensor argument.
+
+from functorch.experimental.control_flow import map
+
+def map_example(xs):
+    def map_fn(x, const):
+        def true_fn(x):
+            return x + const
+        def false_fn(x):
+            return x - const
+        return control_flow.cond(x.sum() > 0, true_fn, false_fn, [x])
+    return control_flow.map(map_fn, xs, torch.tensor([2.0]))
```

Comment:
Is the `map` signature like `functools.partial` in that it can bind additional arguments?
Is this because something like `lamda x: map_fn(x, torch.tensor([2.0])` doesn't work?
Do the additional arguments all have to be tensors?